### ISSUE TITLE
HDFS load cannot load a dataframe that has a byte type column saved

### ISSUE THREAD

**Issue by eemhu** [teragrep/pth_10#187]
**Describe the bug**
```
index=... earliest=...
| teragrep exec tokenizer format bytes input '_raw' output 'tokens'
| teragrep exec hdfs save bloomByteTokens overwrite=true
```
and after
```
| teragrep exec hdfs load bloomByteTokens
```
results in 
```
org.apache.spark.sql.avro.IncompatibleSchemaException: Cannot convert Avro to catalyst because schema at path HEX746f6b656e73 is not compatible (avroType = "int", sqlType = ByteType).
```
Seems that save converts Byte to Int, but load cannot do the same from Int to Byte.
**Expected behavior**
Should load the dataframe successfully.
 
**How to reproduce**
Run the queries described above.
 
**Software version**
pth-10 4.16.0

**Comment by eemhu** [comment]
Perhaps the byte column should be converted to Integer or String before the save and then back to byte format on load.

**Comment by kortemik** [comment]
https://avro.apache.org/docs/1.11.1/specification/#primitive-types 

according to the specification there should be native bytes support within avro. perhaps the List<byte[]> structure within the column is problematic as it is a nested data type.

**Comment by eemhu** [comment]
Converting the ArrayType(ArrayType(ByteType)) field to ArrayType(ArrayType(IntegerType)) seems to fix the issue and allow the dataframe to be loaded successfully - for some reason the Avro writer converts it to integer and the schema expects it to be byte type.

**Comment by eemhu** [comment]
https://spark.apache.org/docs/latest/sql-data-sources-avro.html

It seems that a separate `avroSchema` can be specified - perhaps it can be used to force ByteType?

**Comment by eemhu** [comment]
`org.apache.spark.sql.avro.IncompatibleSchemaException: Cannot convert Catalyst type ByteType to Avro type "bytes".`
Spark `ByteType` is not compatible with the `bytes` Avro type, tested by making a custom AvroSchema where `byteType` is mapped to the avro type `bytes`. 

Think the best option would be to change the tokenizer udf to accept another type such as Integer which can be converted back to byte.

**Comment by eemhu** [comment]
Changed TokenizerUDF return type from ArrayType(ArrayType(ByteType)) to ArrayType(BinaryType), which is natively supported in Avro. BinaryType is the equivalent of Array[Byte] in Scala (byte[] in Java).

**Comment by eemhu** [comment]
internal PR merged, dpf_03 10.0.1 released
